### DISCUSSION TITLE
Variable input size

### DISCUSSION THREAD

**Discussion by kahst** [rdz-oss/BattyBirdNET-Pi#17] [Ideas]
Hi Richard, thanks so much for your amazing work! 

I've been looking at the repo and trying to think how we as the BirdNET team can improve our models to support bat monitoring. One low-hanging fruit that came to mind is variable input size to remove the 3-second limitation. It seems that we can export our base model to allow dynamic resizing at runtime. But I wanted to see if you had any other ideas or feature requests that we could look into.

Best,
Stefan

**Comment by rdz-oss** [comment]
Hi Stefan,

cool! Thank you for your interest. I had been traveling a few days so my late reply.

Congratulations on your success with BirdNET. It has inspired me to get a system working for bats!

I have a few immediate thoughts
- Removing the 3 second input size limitation would indeed be very useful as it makes more combinations of sampling rate and time possible. This would open up space for finding optimal combinations.
- The server.py from BirdNET-Analyzer has become incompatible with the output format expected by BirdNET-PI - on which I have based a bridge to the server.py on BattyBirdNET-Pi. Perhaps we could align on a format that works for you as well as BattyBirdNET-Pi.
- In general I tried to keep BattyBirdNET-Analyzer compatibel with the BirdNET-Analyzer. I have not as yet prioritized integrating the changes made this year on BirdNET-Analyzer, however.
- Evaluation metrics. I was concentrating on proof of concept work and have not as yet benchmarked the detection accuracy. Perhaps you could point me into the right direction of how you do this for systems trained with BirdNET-Analyzer?
- Noise engineering. There appears to be a trade-off between robustness to noise triggers (rain, wind, crickets, cars, ...) and the recall of bat calls, ie precision and loss get worse. As the system is supposed to run 24/7 or at least all night, it should be fairly robust. Perhaps we could exchange on how you balance this for BirdNET?
- I did think about making a new classifier for bat calls. As it is working with the transfer learning from BirdNET that did not turn into a priority as yet.

I will send you a short 'hello note' via my private email so we have that channel, too.

Best,
Richard
